internal/app/triggermessageprocessor.go

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2021 One Track Consulting
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package app
+
+import (
+	"fmt"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/appfunction"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/common"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/runtime"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/trigger"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/messaging"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+	"github.com/hashicorp/go-multierror"
+	"sync"
+)
+
+type simpleTriggerServiceBinding struct {
+	*Service
+	*runtime.GolangRuntime
+}
+
+func (b *simpleTriggerServiceBinding) SecretProvider() messaging.SecretDataProvider {
+	return container.SecretProviderFrom(b.Service.dic.Get)
+}
+
+func NewTriggerServiceBinding(svc *Service) trigger.ServiceBinding {
+	return &simpleTriggerServiceBinding{
+		svc,
+		svc.runtime,
+	}
+}
+
+func (b *simpleTriggerServiceBinding) DIC() *di.Container {
+	return b.Service.dic
+}
+
+func (b *simpleTriggerServiceBinding) BuildContext(env types.MessageEnvelope) interfaces.AppFunctionContext {
+	return appfunction.NewContext(env.CorrelationID, b.Service.dic, env.ContentType)
+}
+
+func (b *simpleTriggerServiceBinding) Config() *common.ConfigurationStruct {
+	return b.config
```

Comment:
should this be
```suggestion
	return b.Service.config
```

Reply:
this finds it because of the way struct composition works but I think its good to be explicit.

Reply:
Got it! Didn't pay enough attention that you used composition.